### PR TITLE
Update ListingTable to use StatisticsConverter

### LINKED ISSUE #10923: Update ListingTable to use `StatisticsConverter`
### Is your feature request related to a problem or challenge?

Thanks to https://github.com/apache/datafusion/issues/10453, we now have the very nice `StatisticsConverter` API ([code link](https://github.com/apache/datafusion/blob/9ab597b2511b23dc951c9f35d00d38c2071e9fea/datafusion/core/src/datasource/physical_plan/parquet/statistics.rs#L566-L589)) that extracts statistics from parquet row groups and is well tested. 

However, there is a different code path that extracts and summarizes statistics in `ListingTable` used for pruning files https://github.com/apache/datafusion/blob/9ab597b2511b23dc951c9f35d00d38c2071e9fea/datafusion/core/src/datasource/file_format/parquet.rs#L310-L373

In addition to being more code, this also seems like it doesn't properly convert the data types

Which is used https://github.com/apache/datafusion/blob/ea21b08e477cc48f458917c132f79c4980c957c1/datafusion/core/src/datasource/file_format/parquet.rs#L511-L535

- @NGA-TRAN  recently made this API public in https://github.com/apache/datafusion/pull/10880 but I think we can do better

### Describe the solution you'd like

I would like to update the code to use `StatisticsConverter` and rather than converting row/column at a time

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by xinlifoobar**
## Which issue does this PR close?

Closes #10923 
Closes https://github.com/apache/datafusion/pull/10924

### CODE REVIEW

File: datafusion/core/src/datasource/file_format/parquet.rs [hunk 1]
```diff
@@ -1449,8 +1420,14 @@ mod tests {
         // column c1
         let c1_stats = &stats.column_statistics[0];
         assert_eq!(c1_stats.null_count, Precision::Exact(1));
-        assert_eq!(c1_stats.max_value, Precision::Absent);
-        assert_eq!(c1_stats.min_value, Precision::Absent);
```
Comment (xinlifoobar) [inline datafusion/core/src/datasource/file_format/parquet.rs:1453]: The test behaviour changes due to Utf8 is not supported in

https://github.com/apache/datafusion/blob/ea21b08e477cc48f458917c132f79c4980c957c1/datafusion/core/src/datasource/file_format/parquet.rs#L298-L376

File: datafusion/core/src/datasource/physical_plan/parquet/statistics.rs [hunk 2]
```diff
@@ -873,7 +921,7 @@ impl<'a> StatisticsConverter<'a> {
 
         let mut builder = UInt64Array::builder(10);
         for metadata in metadatas.into_iter() {
-            let row_count = metadata.num_rows();
+            let row_count = metadata.total_byte_size();
```
Comment (xinlifoobar) [inline datafusion/core/src/datasource/physical_plan/parquet/statistics.rs:924]: Don't trust this diff.

File: datafusion/core/src/datasource/file_format/parquet.rs (hunk 1 above)
Comment (alamb) [reply datafusion/core/src/datasource/file_format/parquet.rs:1453]: It is an improvement@

File: datafusion/core/src/datasource/file_format/parquet.rs [hunk 3]
```diff
@@ -467,88 +385,115 @@ async fn fetch_statistics(
...
+            .fields()
+            .iter()
+            .enumerate()
+            .for_each(|(idx, field)| {
+                match StatisticsConverter::try_new(
+                    field.name(),
+                    &file_schema,
+                    file_metadata.schema_descr(),
+                ) {
+                    Ok(stats_converter) => {
```
Comment (alamb) [inline datafusion/core/src/datasource/file_format/parquet.rs:436]: this is a very nice formulation

**Review by alamb — APPROVED** [review]
Thank you @xinlifoobar  -- this looks like a great improvement. I really like how you structured the code

**Pull request merged**